src/debug.rs

```
@@ -0,0 +1,352 @@
+
+            let call_traces = if only_top {
+                vec![]
+            } else {
+                Arc::try_unwrap(call_tracer_result)
+                    .unwrap()
+                    .take()
+                    .unwrap_or_default()
+            };
+
+            let calltype = if l2_tx.recipient_account() == CONTRACT_DEPLOYER_ADDRESS {
+                DebugCallType::Create
+            } else {
+                DebugCallType::Call
+            };
+
+            let result = match &tx_result.result {
+                ExecutionResult::Success { output } => DebugCall {
+                    gas_used: tx_result.statistics.gas_used.into(),
+                    output: output.clone().into(),
+                    r#type: calltype,
+                    from: l2_tx.initiator_account(),
+                    to: l2_tx.recipient_account(),
+                    gas: l2_tx.common_data.fee.gas_limit,
+                    value: l2_tx.execute.value,
+                    input: l2_tx.execute.calldata().into(),
+                    error: None,
+                    revert_reason: None,
+                    calls: call_traces.into_iter().map(Into::into).collect(),
+                },
+                ExecutionResult::Revert { output } => DebugCall {
+                    gas_used: tx_result.statistics.gas_used.into(),
+                    output: output.encoded_data().into(),
+                    r#type: calltype,
+                    from: l2_tx.initiator_account(),
+                    to: l2_tx.recipient_account(),
+                    gas: l2_tx.common_data.fee.gas_limit,
+                    value: l2_tx.execute.value,
+                    input: l2_tx.execute.calldata().into(),
+                    error: Some(output.to_string()),
+                    revert_reason: Some(output.to_string()),
+                    calls: call_traces.into_iter().map(Into::into).collect(),
+                },
+                ExecutionResult::Halt { reason } => DebugCall {
```

I think for `Halt` we should return an Error: https://github.com/matter-labs/zksync-era/blob/main/core/lib/zksync_core/src/api_server/web3/namespaces/debug.rs#L178-L183
```rust
return Err(Web3Error::SubmitTransactionError(
                    reason.to_string(),
                    vec![],
                ))
```